### PR TITLE
Add ctests.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

**Comment by floitsch** [comment]
No change except for copyright header. Otherwise directly copied from internal repo.

**Comment by floitsch** [comment]
Needed to disable automatic generation of the ctest files with `build_tools`. (Was failing on Windows where the tests weren't compiling.)
Conceptually it feels also cleaner not to compile them with `build_tools`.

File: tests/ctest/ar_generator.cc [hunk 1]
```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2020 Toitware ApS. All rights reserved.
```
Comment (kasperl) [inline tests/ctest/ar_generator.cc:1]: License for these files should be 0BSD?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tests/ctest/ar_generator.cc (hunk 1 above)
Comment (floitsch) [reply tests/ctest/ar_generator.cc:1]: I will change all the licenses in a later PR.

**Pull request merged**